socorro/lib/transaction.py

```diff
@@ -81,24 +75,21 @@ def retry(connection_context, quit_check, fun, **kwargs):
     :returns: varies
 
     """
-    last_failure = None
-
-    for backoff_time in BACKOFF_TIMES:
+    max_attempts = len(BACKOFF_TIMES) - 1
+    for retry_round, backoff_time in enumerate(BACKOFF_TIMES):
         try:
             with connection_context() as conn:
                 return fun(conn, **kwargs)
         except Exception as exc:
-            last_failure = sys.exc_info()
+            if retry_round == max_attempts:
+                # If the last attempt failed, then raise the error.
```

Comment:
hmm I think "last attempt" could imply "previous attempt", changing to "final attempt"